src/upnp.cpp

```diff
@@ -1115,55 +1115,40 @@ void upnp::disable(error_code const& ec)
 	m_socket.close();
 }
 
-namespace
+TORRENT_EXTRA_EXPORT void find_error_code(int type, char const* string, int str_len, error_code_parse_state& state)
 {
-	struct error_code_parse_state
+	if (state.exit) return;
+	if (type == xml_start_tag && !std::strncmp("errorCode", string, size_t(str_len)))
 	{
-		error_code_parse_state(): in_error_code(false), exit(false), error_code(-1) {}
-		bool in_error_code;
-		bool exit;
-		int error_code;
-	};
-
-	void find_error_code(int type, char const* string, int str_len, error_code_parse_state& state)
+		state.in_error_code = true;
+	}
+	else if (type == xml_string && state.in_error_code)
 	{
-		if (state.exit) return;
-		if (type == xml_start_tag && !std::strncmp("errorCode", string, size_t(str_len)))
-		{
-			state.in_error_code = true;
-		}
-		else if (type == xml_string && state.in_error_code)
-		{
-			std::string error_code_str(string, str_len);
-			state.error_code = std::atoi(error_code_str.c_str());
-			state.exit = true;
-		}
+		std::string error_code_str(string, str_len);
+		state.error_code = std::atoi(error_code_str.c_str());
+		state.exit = true;
 	}
+}
 
-	struct ip_address_parse_state: public error_code_parse_state
-	{
-		ip_address_parse_state(): in_ip_address(false) {}
-		bool in_ip_address;
-		std::string ip_address;
-	};
+TORRENT_EXTRA_EXPORT void find_ip_address(int type, char const* string, int str_len, ip_address_parse_state& state)
```

Comment:
I believe ``TORRENT_EXTRA_EXPORT`` is redundant here, and in fact adds the risk of a conflict with the header file